kafka/kafka.go

```diff
@@ -247,58 +253,71 @@ func (cm *consumerMessage) DiscardPayload() {
 	cm.cm = nil
 }
 
-func (ams *asyncMessageSource) ConsumeMessages(ctx context.Context, messages chan<- substrate.Message, acks <-chan substrate.Message) error {
-
-	c, err := cluster.NewConsumerFromClient(ams.client, ams.consumerGroup, []string{ams.topic})
-	if err != nil {
-		return err
+func (cm *consumerMessage) ack() {
+	if cm.cm != nil {
+		cm.sess.MarkMessage(cm.cm, "")
+	} else {
+		off := cm.offset
+		cm.sess.MarkOffset(off.topic, off.partition, off.offset, "")
 	}
```

Comment:
Nitpick: could we format this to avoid the else?
```go
if cm.cm != nil {
	cm.sess.MarkMessage(cm.cm, "")
	return
}

off := cm.offset
cm.sess.MarkOffset(off.topic, off.partition, off.offset, "")
```